x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlExecutionInfo.java

```diff
@@ -478,7 +478,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             {
                 builder.field(STATUS_FIELD.getPreferredName(), getStatus().toString());
                 builder.field(INDICES_FIELD.getPreferredName(), indexExpression);
-                if (took != null) {
+                if (took != null && status != Status.RUNNING) {
```

Comment:
Question: if the query didn't finish yet, it won't report anything here at all, do I understand correct? If so, maybe it makes sense to report how long it has been running? I added a function `tookSoFar()` here: https://github.com/elastic/elasticsearch/pull/116365/files#diff-48d6ae7f9f13d31f768405be81432badab35517119f3c4e438892e7ff5c0d7a0R166-R173 maybe using the same thing makes sense?

Reply:
I added this because in my testing I saw a race condition where the overall took time had been set but the running flag had not yet been set.

But in _search, took time is shown as the search progresses and has not yet finished, so maybe we could do the same in ES|QL?  _async_search also has additional fields that ESQL async query does not:

```
  "start_time_in_millis": 1732716957522,
  "expiration_time_in_millis": 1733148957522,
  "completion_time_in_millis": 1732716984620,
  "response": {
    "took": 27098,
```
That completion time is only set at the very end. I'd say we should ask Product and Kibana these questions - do they want `took` only at the end or for it to mean "took so far"? And do they want the start and completion timestamps?

Reply:
Actually the above comment, while correct belongs on the overall took time section. I did ask Kibana what they want and they recommended to do it like search - have the took time always present so I added your tookSoFar method to EsqlExecutionInfo.

In terms of this code (line 481), this is at the Cluster object level. There I don't want to expose intermediate took times, since I use that value in ComputeListener to update the value each time another data node finishes, so it really isn't a "took so far" thing and would be redundant to the overall took-so-far. Instead, like _search, it should show per-cluster took times only once it is finished.